examples/protocol_specification_ex/output/two_party_negotiation/message.py

```diff
@@ -23,14 +38,57 @@ def __eq__(self, other):
             return False
 
 
+class Ali:
+    """This class represents an instance of Ali."""
+
+    def __init__(self):
+        """Initialise an instance of Ali."""
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        """Compare two Ali instances."""
+        if type(other) is type(self):
+            raise NotImplementedError
+        else:
+            return False
+
+
+class _IOTApp:
+    """This class represents an instance of _IOTApp."""
+
+    def __init__(self):
+        """Initialise an instance of _IOTApp."""
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        """Compare two _IOTApp instances."""
+        if type(other) is type(self):
+            raise NotImplementedError
+        else:
+            return False
+
+
 class TwoPartyNegotiationMessage(Message):
     """A protocol for negotiation over a fixed set of resources involving two parties."""
 
     protocol_id = "two_party_negotiation"
 
     _speech_acts = {
```

Comment:
I don't see why this variable needs to be part of the class. It's only used for its generation so can be dropped.

Reply:
Change to set of performatives.

Reply:
I remember why we added `_speech_acts`.

It was needed for the shorter version of the code that checks the contents of a message via a loop that goes thought the content definitions in the `_speech_act` and checks the contents of the message match those in the definition in terms of name, number and type.

We could delete the `_speech_act` variable, but if we ever want to simplify the code, we need to put it back in.